pkg/result/processors/fixer.go

```diff
@@ -43,218 +53,221 @@ func (p Fixer) Process(issues []result.Issue) ([]result.Issue, error) {
+				name, x, y := linter, edits[invalid-1], edits[invalid]
+				excludedLinters[name] = struct{}{}
+
+				err := diff3Conflict(path, name, name, []diff.Edit{x}, []diff.Edit{y})
+				// TODO(ldez) TUI?
+				p.log.Warnf("Changes related to %q are skipped for the file %q: %v",
+					name, path, err)
+			}
 		}
-	}
 
-	issues = p.findNotIntersectingIssues(issues)
+		// Does any pair of different linters create edits that conflict?
+		for j := range linters {
+			for k := range linters[:j] {
+				x, y := linters[j], linters[k]
+				if x > y {
+					x, y = y, x
+				}
 
-	if err = p.writeFixedFile(origFileLines, issues, tmpOutFile); err != nil {
-		tmpOutFile.Close()
-		_ = robustio.RemoveAll(tmpOutFile.Name())
-		return err
-	}
+				_, foundX := excludedLinters[x]
+				_, foundY := excludedLinters[y]
+				if foundX || foundY {
+					continue
+				}
 
-	tmpOutFile.Close()
+				xedits, yedits := linterToEdits[x], linterToEdits[y]
 
-	if err = robustio.Rename(tmpOutFile.Name(), filePath); err != nil {
-		_ = robustio.RemoveAll(tmpOutFile.Name())
-		return fmt.Errorf("failed to rename %s -> %s: %w", tmpOutFile.Name(), filePath, err)
-	}
+				combined := slices.Concat(xedits, yedits)
 
-	return nil
-}
+				if _, invalid := validateEdits(combined); invalid > 0 {
+					excludedLinters[x] = struct{}{}
+					p.log.Warnf("Changes related to %q are skipped for the file %q due to conflicts with %q.", x, path, y)
+				}
+			}
+		}
 
-func (p Fixer) mergeLineIssues(lineNum int, lineIssues []result.Issue, origFileLines [][]byte) *result.Issue {
-	origLine := origFileLines[lineNum-1] // lineNum is 1-based
+		var edits []diff.Edit
+		for linter := range linterToEdits {
+			if _, found := excludedLinters[linter]; !found {
+				edits = append(edits, linterToEdits[linter]...)
+			}
+		}
 
-	if len(lineIssues) == 1 && lineIssues[0].Replacement.Inline == nil {
-		return &lineIssues[0]
+		editsByPath[path], _ = validateEdits(edits) // remove duplicates. already validated.
 	}
 
-	// check issues first
-	for ind := range lineIssues {
-		li := &lineIssues[ind]
+	var editError error
 
-		if li.LineRange != nil {
-			p.log.Infof("Line %d has multiple issues but at least one of them is ranged: %#v", lineNum, lineIssues)
-			return &lineIssues[0]
+	// Now we've got a set of valid edits for each file. Apply them.
+	for path, edits := range editsByPath {
+		contents, err := p.fileCache.GetFileBytes(path)
+		if err != nil {
+			editError = errors.Join(editError, err)
+			continue
 		}
 
-		inline := li.Replacement.Inline
+		out, err := diff.ApplyBytes(contents, edits)
+		if err != nil {
+			editError = errors.Join(editError, err)
+			continue
+		}
 
-		if inline == nil || len(li.Replacement.NewLines) != 0 || li.Replacement.NeedOnlyDelete {
-			p.log.Infof("Line %d has multiple issues but at least one of them isn't inline: %#v", lineNum, lineIssues)
-			return li
+		// Try to format the file.
+		if formatted, err := format.Source(out); err == nil {
```

Comment:
Nice! I don't think we actually formatted the source after fixing before but this will make suggesting fixes even easier (e.g. indentation and multiple spaces etc. isn't as important).